src/interface.jl

```diff
@@ -606,10 +606,10 @@ coordinates(obj) = coordinates(trait(obj), obj)
 """
     convert(type::CustomGeom, geom)
 
-Convert `geom` into the `CustomGeom` type if both geom as the CustomGeom package
-have implemented GeoInterface.
+Create a `CustomGeom` from any `geom` that implements the GeoInterface.
 """
 convert(T, geom) = convert(T, geomtrait(geom), geom)
```

Comment:
If this was changed to `Base.convert`, it would be highly problematic -- but not just because of potential ambiguities this introduces, but also because it would invalidate basically every method in the system, forcing Julia to recompile tons of stuff. So basically after `using GeoInterface` everything would have the "first time invocation overhead" again. That'd be really bad, esp. now that Julia 1.9 or 1.10 will add native code caching (i.e. generated machine code will be cached as part of precompilation -- speeding up TTFX quite a bit -- unless packages kill the benefit by invalidating methods...).

Reply:
Noone suggested that  you may misunderstand this PR.

If we invalidate anything with `convert` it would be with type pyracy, which is much higher up the list of "things not to do".

But we are all on board with using the package `convert` it's just what to do with the fact that we already have implementations of `Base.convert` in the wild.